go/obscuronode/enclave/enclave.go

```diff
@@ -183,21 +191,49 @@ func (e *enclaveImpl) ProduceGenesis() BlockSubmissionResponse {
 	}
 }
 
-func (e *enclaveImpl) IngestBlocks(blocks []*types.Block) {
-	for _, block := range blocks {
+func (e *enclaveImpl) IngestBlocks(blocks []*types.Block) []BlockSubmissionResponse {
+	result := make([]BlockSubmissionResponse, len(blocks))
+	for i, block := range blocks {
 		e.db.StoreBlock(block)
-		updateState(block, e.db, e.blockResolver)
+		bs := updateState(block, e.db, e.blockResolver)
+		if bs == nil {
+			result[i] = BlockSubmissionResponse{
+				L1Hash:            block.Hash(),
+				L1Height:          e.blockResolver.HeightBlock(block),
+				L1Parent:          block.ParentHash(),
+				IngestedBlock:     true,
+				IngestedNewRollup: false,
+			}
+		} else {
+			var rollup nodecommon.ExtRollup
+			if bs.foundNewRollup {
+				rollup = bs.Head.ToExtRollup()
+			}
+			result[i] = BlockSubmissionResponse{
+				L1Hash:            bs.Block.Hash(),
+				L1Height:          e.blockResolver.HeightBlock(bs.Block),
+				L1Parent:          bs.Block.ParentHash(),
+				L2Hash:            bs.Head.Hash(),
+				L2Height:          e.db.HeightRollup(bs.Head),
+				L2Parent:          bs.Head.Header.ParentHash,
+				ProducedRollup:    rollup,
+				IngestedBlock:     true,
+				IngestedNewRollup: bs.foundNewRollup,
+			}
+		}
 	}
+
+	return result
 }
 
 func (e *enclaveImpl) SubmitBlock(block types.Block) BlockSubmissionResponse {
 	// Todo - investigate further why this is needed.
 	// So far this seems to recover correctly
-	defer func() {
```

Comment:
Can we just get rid of this commented out code and the todo?

Reply:
The point of this was to see if a node successfully recovered after a race which paniced.
I'll get rid of it, since it's swallowing normal panics and it makes debugging harder

Reply:
not sure what you mean. I don't see it

Reply:
<img width="922" alt="Screenshot 2022-03-23 at 10 49 49" src="https://user-images.githubusercontent.com/24230167/159682969-43d01bcd-4cc6-4069-bc98-a675a64a5331.png">